diff -rupN /usr/lib/python3.4/site-packages/chartit/chartdata.py ./chartdata.py
--- /usr/lib/python3.4/site-packages/chartit/chartdata.py	2015-10-06 07:31:14.000000000 +0200
+++ ./chartdata.py	2015-10-05 18:40:47.484444012 +0200
@@ -125,7 +125,14 @@ class DataPool(object):
         for tk_td_tuples in self.query_groups:
             src = tk_td_tuples[0][1]['source']
             vqs = src.values(*(td['field'] for (tk, td) in tk_td_tuples))
-            yield tk_td_tuples, vqs
+            vqs2 = []
+            for v in vqs:
+                for (tk, td) in tk_td_tuples:
+                    f = td.get('fn')
+                    if(f):
+                        v[td['field']] = f(v[td['field']])
+                vqs2.append(v)
+            yield tk_td_tuples, vqs2
     
     def _get_data(self):
         for tk_td_tuples, vqs in self._generate_vqs():
@@ -565,4 +572,4 @@ class PivotDataPool(DataPool):
                 self.cv = [self.mapf(cv) for cv in self.cv_raw]
                 if self.mts: 
                     combined = sorted(zip(self.cv, self.cv_raw),key=self.sortf)
-                    self.cv, self.cv_raw = zip(*combined)
\ No newline at end of file
+                    self.cv, self.cv_raw = zip(*combined)
Binary files /usr/lib/python3.4/site-packages/chartit/chartdata.pyc and ./chartdata.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/charts.pyc and ./charts.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/exceptions.pyc and ./exceptions.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/highcharts/hcoptions.pyc and ./highcharts/hcoptions.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/highcharts/__init__.pyc and ./highcharts/__init__.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/highcharts/__pycache__/hcoptions.cpython-34.pyc and ./highcharts/__pycache__/hcoptions.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/highcharts/__pycache__/__init__.cpython-34.pyc and ./highcharts/__pycache__/__init__.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__init__.pyc and ./__init__.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/models.pyc and ./models.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__pycache__/exceptions.cpython-34.pyc and ./__pycache__/exceptions.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__pycache__/__init__.cpython-34.pyc and ./__pycache__/__init__.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__pycache__/models.cpython-34.pyc and ./__pycache__/models.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__pycache__/utils.cpython-34.pyc and ./__pycache__/utils.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/__pycache__/validation.cpython-34.pyc and ./__pycache__/validation.cpython-34.pyc differ
diff -rupN /usr/lib/python3.4/site-packages/chartit/templatetags/chartit.py ./templatetags/chartit.py
--- /usr/lib/python3.4/site-packages/chartit/templatetags/chartit.py	2015-10-06 07:31:14.000000000 +0200
+++ ./templatetags/chartit.py	2015-10-06 07:32:44.067403998 +0200
@@ -1,7 +1,7 @@
 from itertools import izip_longest
 
 from django import template
-from django.utils import simplejson
+import simplejson
 from django.utils.safestring import mark_safe
 from django.conf import settings
 import posixpath
@@ -71,4 +71,4 @@ def load_charts(chart_list=None, render_
                                         CHART_LOADER_URL))
     else:
         embed_script = embed_script %((), CHART_LOADER_URL)
-    return mark_safe(embed_script)
\ No newline at end of file
+    return mark_safe(embed_script)
Binary files /usr/lib/python3.4/site-packages/chartit/templatetags/chartit.pyc and ./templatetags/chartit.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/templatetags/__init__.pyc and ./templatetags/__init__.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/templatetags/__pycache__/chartit.cpython-34.pyc and ./templatetags/__pycache__/chartit.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/templatetags/__pycache__/__init__.cpython-34.pyc and ./templatetags/__pycache__/__init__.cpython-34.pyc differ
Binary files /usr/lib/python3.4/site-packages/chartit/utils.pyc and ./utils.pyc differ
diff -rupN /usr/lib/python3.4/site-packages/chartit/validation.py ./validation.py
--- /usr/lib/python3.4/site-packages/chartit/validation.py	2015-10-06 07:31:14.000000000 +0200
+++ ./validation.py	2015-10-05 18:51:28.581047597 +0200
@@ -236,6 +236,12 @@ def _convert_dps_to_dict(series_list):
                         else:
                             raise APIInputError("Expecting a basestring or "
                                                 "dict in place of: %s" %tv)
+                elif isinstance(term,tuple):
+                    t,fn = term
+                    opt = copy.deepcopy(options)
+                    opt['fn'] = fn
+                    series_dict[t] = opt 
+
         elif isinstance(terms, dict):
             for tk, tv in terms.items():
                 if isinstance(tv, basestring):
Binary files /usr/lib/python3.4/site-packages/chartit/validation.pyc and ./validation.pyc differ
